### PR TITLE
AIRFLOW-4912: Support to update nested keys in emr_defaults conn extras.

### PR DESCRIPTION
**Pull request by jyoti-arora1991**
Added support to update emr_defaults extra's in airflow dag using job_flow_overrides.

### Jira

- [x] My PR addresses the following [AIRFLOW-4912](https://issues.apache.org/jira/browse/AIRFLOW-4912/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
tests.utils:TestHelpers:test_update_dictionary_recursively:
### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

**Review by ashb — CHANGES_REQUESTED** [review]
I'd rather we left the format exactly as the AWS API accepts (i.e. nested, not dot separated) and then we can use something like https://pypi.org/project/deepmerge/ to handle the merging.

File: tests/contrib/hooks/test_emr_hook.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def test_create_job_flow_extra_args(self):
         # coverage of EMR isn't 100% it turns out.
         cluster = hook.create_job_flow({'Name': 'test_cluster',
                                         'ReleaseLabel': '',
-                                        'AmiVersion': '3.2'})
+                                        'AmiVersion': '3.2', 'Instances': {'TerminationProtected': True}})
```
Comment (ashb) [inline tests/contrib/hooks/test_emr_hook.py]: Could you extend the tests here to make sure that we are actually getting a merge here

File: tests/utils/test_helpers.py [hunk 2]
```diff
@@ -52,6 +52,7 @@ def signal_handler(signum, frame):
     def _parent_of_ignores_sigterm(parent_pid, child_pid, setup_done):
         def signal_handler(signum, frame):
             pass
+
```
Comment (ashb) [inline tests/utils/test_helpers.py]: Could you revert the changes to this file please?

Comment (jyoti-arora1991) [reply tests/utils/test_helpers.py]: Changes incorporated

**Comment by OmerJog** [comment]
@ashb is there something else needed here other than rebase?

**Review by ashb — APPROVED** [review]
Rebase please @jyoti-arora1991

**Comment by OmerJog** [comment]
@jyoti-arora1991 any progress with rebasing?
